packages/mobile/src/screens/search-screen-v2/screens/FilterBpmScreen.tsx

```diff
@@ -75,22 +75,27 @@ const rangeOptions = [
 ]
 
 const BpmRangeView = ({ value, setValue }: ViewProps) => {
-  const [bpmRange, setBpmRange] = useState<string>()
-  const [minBpm, setMinBpm] = useState('')
-  const [maxBpm, setMaxBpm] = useState('')
+  const isValueRangeOption =
+    value?.includes('-') &&
+    Boolean(rangeOptions.find((opt) => opt.value === value))
+  const minMaxValue = value?.includes('-') ? value.split('-') : null
+
+  const [bpmRange, setBpmRange] = useState<string>(
+    isValueRangeOption ? (value as string) : minMaxValue ? 'custom' : ''
```

Comment:
It took me a sec to parse this but I'm just not a fan of nested ternaries 🙈

Reply:
yea i don't like it either, i'll probably rewrite this